editor/store/effects.js

```diff
@@ -254,6 +253,8 @@ export default {
 		dispatch( removeNotice( TRASH_POST_NOTICE_ID ) );
 		apiRequest( { path: `/wp/v2/${ basePath }/${ postId }`, method: 'DELETE' } ).then(
 			() => {
+				const post = getCurrentPost( getState() );
+				dispatch( resetPost( { ...post, status: 'trash' } ) );
 				dispatch( {
 					...action,
 					type: 'TRASH_POST_SUCCESS',
```

Comment:
This action type is never referenced elsewhere. Should we remove it? My only hesitation might be is that it's complementing the failing type, which is in-fact referenced. The counter-point may be that the existence of this action type may be misinterpreted as a hint that it has resulting handling. I'd lean toward removal.

Aside: I'd also like to refactor the notices behavior, which would include consolidating the `TRASH_POST_FAILURE` action type.